ic-cdk/src/call.rs

```diff
@@ -432,6 +432,91 @@ pub struct CandidDecodeFailed {
     candid_error: String,
 }
 
+/// Extension trait for error types to provide additional methods.
+pub trait CallErrorExt {
+    /// Checks if the error is a clean reject.
+    /// A clean reject means that there must be no state changes on the callee side.
+    fn is_clean_reject(&self) -> bool;
+    /// Checks if the failed call is immediately retryable.
```

Comment:
I'd suggest:
/// Checks if the failed call is immediately retryable. 
/// where "immediately" means in the update method that's
/// handling the error, as opposed to a background timer
/// or heartbeat.